### PR TITLE
feat(image-hotspots): add minimapBehavior prop for better control

### LINKED ISSUE #1356: [imagecard stories] Auto-hiding of minimap should be an option, not the only behavior.
### Describe the bug

- Originally the design shows the minimap 'showing' at all times. 
- There is currently a knob that allows you to specify that the minimap should not show ever (although this seems to be broken).
- Monitor team got valid feedback that in their usecases where the image is generally small, it's beneficial to have the minimap hide automatically when not actively panning. This was then made the default behavior in Monitor, making it impossible to have the original intent (always show minimap). However it also changed the default behavior of the underlying PAL component unilaterally.
- There are usecases that still exist of images at larger sizes that benefit from having the minimap as a way to navigate around it, which is impossible when it is auto-hidden as it cannot be interacted with.

- This bug is to rectify that change.

### To Reproduce

image

1. Go to any of the image hotspot stories (not sure on what the difference is here from image card... 
2. Click on the image and drag to see the minimap.
3. Scroll down to the knobs and click the hide minimap.
4. See error in that there is no way to always show the minimap (when zoomed), nor does the hide minimap knob seem to function, as it will still show on pan even when this is checked.
image

### Expected behavior

- The current 'auto hide minimap' behavior added as an option to its behavior like the 'hide' option already.
- Optimum would be the make the behavior a choice: 
  1. Always show
  2. Show on pan
  3. Always hide

### Specific timeline issues / requests

No specific timeline identified currently.

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #1356 

**Summary**

- adds a `minimapBehavior` prop for better control over when the minimap is displayed.

**Change List (commits, features, bugs, etc)**

- Add `minimapBehavior` prop
- Deprecate `hideMinimap` prop
- Add tests to confirm behavior is as expected
- fixed styles on minimize icon when zoomed in.
- updated snapshots

**Acceptance Test (how to verify the PR)**

- in the [https://deploy-preview-2345--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-imagehotspots--landscape-image-landscape-container story](https://deploy-preview-2345--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-imagehotspots--landscape-image-landscape-container)
  - zooming in and panning should show the minimap by default.
  - change the minimapBehavior prop to 'hide' and the minimap should always be hidden. even when panning.
  - change the minimapBehavior prop to 'show' and the minimap should always be visible--even when at default zoom or when panning.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- previously expected behavior (only showing minimap when panning) should be seen in all stories

### CODE REVIEW

**Comment by bjornalm** [comment]
If I go to the story  https://deploy-preview-2345--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-imagehotspots--landscape-image-landscape-container story and zoom in, there is no minimap by default like the acceptance test states.

File: packages/react/src/components/ImageCard/ImageHotspots.jsx [hunk 1]
```diff
@@ -29,7 +29,23 @@ const propTypes = {
...
+        return new Error(
+          `${componentName}: '${propName}' prop is deprecated in favor of 'minimapBehavior="${
+            value ? 'hide' : 'show'
+          }"'.`
+        );
+      }
+    }
+  },
+  /* 'hide' always hides the minimap, 'show' always shows the minimap, 'onPan' only shows the minimap when panning an image */
+  minimapBehavior: PropTypes.oneOf(['hide', 'show', 'onPan']),
```
Comment (bjornalm) [inline packages/react/src/components/ImageCard/ImageHotspots.jsx:48]: onPan is a bit ambiguous, do we show or hide on pan? Can we call is showOnPan instead?

**Review by bjornalm — APPROVED** [review]
Nice work, just minor naming issue

File: packages/react/src/components/ImageCard/ImageHotspots.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/ImageCard/ImageHotspots.jsx]: yep. that's more clear.

**Pull request merged**